src/Albany_Application.cpp

```diff
@@ -735,6 +735,71 @@ Application::finalSetUp(
   for (int i = 0; i < responses.size(); ++i) { responses[i]->postRegSetup(); }
 }
 
+template<typename Traits>
+void
+Application::setSideSetExtents(Teuchos::RCP<PHX::FieldManager<PHAL::AlbanyTraits>>& in_fm) const
+{
+  // get number of worksets
+  const auto& wsElNodeEqID = disc->getWsElNodeEqID();
+  unsigned int const numWorksets = wsElNodeEqID.size();
+
+  // compute largest workset size over all worksets for each sideset name
+  std::map<std::string, unsigned int> maxSideSetSizes;
+  for (unsigned int i = 0; i < numWorksets; ++i) { 
+    const LocalSideSetInfoList& sideSetView = disc->getSideSetViews(i);
+    for (auto it = sideSetView.begin(); it != sideSetView.end(); ++it) {
+      if (maxSideSetSizes.find(it->first) == maxSideSetSizes.end()) {
+        maxSideSetSizes[it->first] = 0;
+      }
+      unsigned int sideSetSize = it->second.size;
+      maxSideSetSizes[it->first] = std::max(maxSideSetSizes[it->first], sideSetSize);
+    }
+  }
+
+  // Iterate over tags and set extents for sideset fields
+  const auto& tags = in_fm->getFieldTagsForSizing<Traits>();
+  in_fm->buildDagForType<Traits>();
+  for (auto& t : tags) {
+    auto& t_dl = t->nonConstDataLayout();
+    std::vector<PHX::Device::size_type> t_dims;
+    t_dl.dimensions(t_dims);
+
+    // Check if dimension[0] is Side
+    std::string first_dim_name = t_dl.name(0);
+    std::string t_identifier = t_dl.identifier();
+    std::string sideSetName = t_identifier.substr(0, t_identifier.find("<"));
+
+    if (first_dim_name == "Side") {
+
+      t_dims[0] = maxSideSetSizes[sideSetName];
+
+      switch (t_dims.size()) {
+        case 1:
+          t_dl.setExtents(t_dims[0]);
+          break;
+        case 2:
+          t_dl.setExtents(t_dims[0], t_dims[1]);
+          break;
+        case 3:
+          t_dl.setExtents(t_dims[0], t_dims[1], t_dims[2]);
+          break;
+        case 4:
+          t_dl.setExtents(t_dims[0], t_dims[1], t_dims[2], t_dims[3]);
+          break;
+        case 5:
+          t_dl.setExtents(t_dims[0], t_dims[1], t_dims[2], t_dims[3], t_dims[4]);
+          break;
+        case 6:
+          t_dl.setExtents(t_dims[0], t_dims[1], t_dims[2], t_dims[3], t_dims[4], t_dims[5]);
+          break;
+        default:
+          TEUCHOS_TEST_FOR_EXCEPTION (true, std::logic_error,
+            "Error! Sideset dynamic sizing as encountered a layout with more field tags than expected.\n");
+      }
+    }
```

Comment:
Can we also throw if t_dims[0] is zero here? This should ensure that all layouts used by the field manager have a size that's initialized.